rclpy/src/rclpy/_rclpy.c

```diff
@@ -1599,6 +1599,72 @@ rclpy_publish(PyObject * Py_UNUSED(self), PyObject * args)
   Py_RETURN_NONE;
 }
 
+/// Publish a serialized message
+/**
+ * Raises ValueError if pypublisher is not a publisher capsule
+ * Raises RuntimeError if the message cannot be published
+ *
+ * \param[in] pypublisher Capsule pointing to the publisher
+ * \param[in] pymsg serialized message to send
+ * \return NULL
+ */
+
+static PyObject *
+rclpy_publish_raw(PyObject * Py_UNUSED(self), PyObject * args)
+{
+  PyObject * pypublisher;
+  PyBytesObject * pymsg;
+  char * message;
+  Py_ssize_t pymsg_size;
+
+  if (!PyArg_ParseTuple(args, "OO", &pypublisher, &pymsg)) {
+    return NULL;
+  }
+
+  int err = PyBytes_AsStringAndSize(pymsg, &message, &pymsg_size);
+
+  if (err == -1) {
+    PyErr_Format(RCLError, "Failed to fetch buffer contents: %d", err);
```

Comment:
[PyBytes_AsStringAndSize](https://docs.python.org/3/c-api/bytes.html#c.PyBytes_AsStringAndSize) already raises an exception, so I don't think we should raise another. I think we should remove this line.